backend/src/main/scala/com/softwaremill/adopttapir/starter/StarterModel.scala

```diff
@@ -1,25 +1,43 @@
 package com.softwaremill.adopttapir.starter
 
-case class StarterDetails(
-    tapirVersion: String,
-    scalaVersion: String,
-    sbtVersion: String,
-    projectDetails: ProjectDetails,
-    moduleDependencies: ModuleDependencies
-)
+sealed trait StarterDetails {
+  val projectName: String
+  val groupId: String
+  val serverImplementation: ServerImplementation
+  val tapirVersion: String
+}
+
+object StarterDetails {
+  val defaultTapirVersion = "1.0.0-RC1"
 
-case class ProjectDetails(group: String, artifact: String, projectName: String, packageName: String)
-case class ModuleDependencies(dependencies: List[Dependency])
-case class Dependency(value: String) extends AnyVal
+  case class FutureStarterDetails(
```

Comment:
hm why do we have 3 identical case classes? wouldn't one with an `effectImplementation` field suffice, just as we have `serverImplementation`?

Reply:
I will remove different implementation, as they not add any value comparing to one case class.